gpMgmt/bin/gppylib/test/unit/test_unit_gprebalance.py

```diff
@@ -0,0 +1,170 @@
+from gppylib.system.environment import GpCoordinatorEnvironment
+from gppylib.db import dbconn
+import io
+import sys
+
+
+def initGparrayFromFile(filename):
+    segdbs = []
+    with open(filename, 'r') as fp:
+        for line in fp:
+            if not line.lstrip().startswith('#'):
+                segdbs.append(Segment.initFromString(line))
+    return GpArray(segdbs, segdbs)
+
+
+class GpTestRebalance(GpTestCase):
+    def setUp(self):
+        gprebalance_file = os.path.abspath(
+            os.path.dirname(__file__) + "/../../../gprebalance")
+        self.subject = imp.load_source('gprebalance', gprebalance_file)
+        self.old_sys_argv = sys.argv
+        sys.argv = []
+        self.options, self.args, self.parser = self.subject.parseargs()
+        self.options.mirroring = 'grouped'
+
+        self.subject.logger = Mock(
+            spec=['log', 'warn', 'info', 'debug', 'error', 'warning', 'fatal', 'exception'])
+
+        self.subject.check_running_gputils = Mock(return_value=False)
+
+        self.apply_patches([
+            patch('builtins.open', mock_open(), create=True),
+            patch('builtins.input'),
+            patch('gprebalance.dbconn.DbURL', return_value=Mock(), spec=DbURL),
+            patch('gprebalance.dbconn.connect', return_value=Mock()),
+            patch('gprebalance.GpCoordinatorEnvironment',
+                  return_value=Mock(), spec=GpCoordinatorEnvironment),
+            patch('gprebalance.configurationInterface.getConfigurationProvider'),
+            patch('os.path.exists', return_value=Mock()),
+            patch('gprebalance.get_default_logger',
+                  return_value=self.subject.logger),
+        ])
+        self.input_mock = self.get_mock_from_apply_patch("input")
+        self.getConfigProviderFunctionMock = self.get_mock_from_apply_patch(
+            "getConfigurationProvider")
+        self.gpCoordinatorEnvironmentMock = self.get_mock_from_apply_patch(
+            "GpCoordinatorEnvironment")
+        self.previous_coordinator_data_directory = os.getenv(
+            'COORDINATOR_DATA_DIRECTORY', '')
+        self.previous_coordinator_data_directory = os.getenv(
+            'COORDINATOR_DATA_DIRECTORY', '')
+        os.environ["COORDINATOR_DATA_DIRECTORY"] = '/tmp/dirdoesnotexist'
+        configProviderMock = Mock(spec=GpConfigurationProvider)
+        self.getConfigProviderFunctionMock.return_value = configProviderMock
+        configProviderMock.initializeProvider.return_value = configProviderMock
+        self.gpCoordinatorEnvironmentMock.return_value.getCoordinatorPort.return_value = 123456
+
+    def tearDown(self):
+        os.environ['COORDINATOR_DATA_DIRECTORY'] = self.previous_coordinator_data_directory
+        sys.argv = self.old_sys_argv
+        super(GpTestRebalance, self).tearDown()
+
+    @patch('gprebalance.GpArray.initFromCatalog',
+           return_value=initGparrayFromFile(os.path.dirname(__file__) + "/data/gprebalance/balanced_grouped.array"))
```

Comment:
how about something like
```suggestion
           return_value=initGparrayFromFile("balanced_grouped"))
```
(move repeated common parts into initGparrayFromFile function)

Reply:
done